miden-tx/src/prover/mod.rs

```diff
@@ -93,7 +83,17 @@ impl TransactionProver for LocalTransactionProver {
         &self,
         tx_witness: TransactionWitness,
     ) -> Result<ProvenTransaction, TransactionProverError> {
-        let TransactionWitness { tx_inputs, tx_args, advice_witness } = tx_witness;
+        let TransactionWitness {
+            tx_inputs,
+            tx_args,
+            advice_witness,
+            account_codes,
+        } = tx_witness;
+
+        for account_code in account_codes {
+            // load the code mast forest to the mast store
+            self.mast_store.load_account_code(&account_code);
+        }
```

Comment:
We should construct `account_code_comments` based on the data in `account_codes` and pass it to the host constructor on line 115.